tests/ltp/patches/setsid01.patch

```diff
@@ -0,0 +1,260 @@
++    }
++
++    if (!fail) {
++      tst_resm(TPASS, "all misc tests passed");
++    }
++  }
++  cleanup();
++  tst_exit();
+ 
+-		/*
+-		 * When the process group having forked of a child
+-		 * and then it attached itself to another process
+-		 * group and tries to setsid
+-		 */
+-		pid = FORK_OR_VFORK();
+-
+-		if (pid == 0) {
+-			if ((pid = FORK_OR_VFORK()) == -1) {
+-				tst_resm(TFAIL, "Fork failed");
++}
+ 
+-			}
+-			if (pid == 0) {
+-#ifdef UCLINUX
+-				if (self_exec(argv0, "n", 1) < 0) {
+-					tst_resm(TFAIL, "self_exec failed");
+ 
+-				}
+-#else
+-				do_child_1();
+-#endif
+-			} else {
+-				if (setpgid(0, 0) < 0) {
+-					tst_resm(TFAIL,
+-						 "setpgid(parent) failed: %s",
+-						 strerror(errno));
+-					fail = 1;
+-				}
+-
+-				if ((ret = wait(&status)) > 0) {
+-					if (status != 0) {
+-						tst_resm(TFAIL,
+-							 "Test {%d} exited "
+-							 "status 0x%0x (wanted 0x0)",
+-							 ret, status);
+-						fail = 1;
+-					}
+-				}
+-			}
+-			exit(0);
+-		} else {
+-			if ((ret = wait(&status)) > 0) {
+-				if (status != 0) {
+-					tst_resm(TFAIL, "Test {%d} exited "
+-						 "status 0x%0x (wanted 0x0)",
+-						 ret, status);
+-					fail = 1;
+-				}
+-			}
+-		}
+-
+-		if (!(fail || exno)) {
+-			tst_resm(TPASS, "all misc tests passed");
+-		}
+-	}
+-	cleanup();
+-	tst_exit();
++static void* do_master_thread(void* arg)
++{
++  int s;
++  pthread_t child_tid;
+ 
+-}
++  setpgid(0,0);
+ 
+-/*
+- * do_child_1()
+- */
+-void do_child_1(void)
+-{
+-	int exno = 0;
+-	int retval, ret, status;
+-	int pid;
++  s = pthread_create(&child_tid, NULL, do_child_thread, NULL);
+ 
+-	sleep(1);
++  if (s == 0)
++  {
++    printf("%s(): thread created fine ...\n", __FUNCTION__);
++    pthread_join(child_tid, NULL);
++  }
++  else
++    printf("%s(): thread created error ...\n", __FUNCTION__);
+ 
+-	if (setpgid(0, 0) < 0) {
+-		tst_resm(TFAIL, "setpgid(0,0) failed: %s", strerror(errno));
+-		exno = 1;
+-	}
++  pthread_exit(NULL);
+ 
+-	if ((pid = FORK_OR_VFORK()) == -1) {
+-		tst_brkm(TFAIL, NULL, "Fork failed");
+-	}
+-	if (pid == 0) {
+-#ifdef UCLINUX
+-		if (self_exec(argv0, "n", 2) < 0) {
+-			tst_brkm(TFAIL, NULL, "self_exec failed");
+-		}
+-#else
+-		do_child_2();
+-#endif
+-	} else {
+-		retval = setpgid(0, getppid());
+-		if (retval < 0) {
+-			tst_resm(TFAIL, "setpgid failed, errno :%d", errno);
+-			exno = 2;
+-		}
+-
+-		retval = setsid();
+-
+-		if (errno == EPERM) {
+-			tst_resm(TPASS, "setsid SUCCESS to set "
+-				 "errno to EPERM");
+-		} else {
+-			tst_resm(TFAIL, "setsid failed, expected %d,"
+-				 "return %d", -1, errno);
+-			exno = 3;
+-		}
+-		kill(pid, SIGKILL);
+-		if ((ret = wait(&status)) > 0) {
+-			if (status != 9) {
+-				tst_resm(TFAIL,
+-					 "Test {%d} exited status 0x%-x (wanted 0x9)",
+-					 ret, status);
+-				exno = 4;
+-			}
+-		}
+-	}
+-	exit(exno);
+ }
+ 
+-/*
+- * do_child_2()
+- */
+-void do_child_2(void)
++
++static void* do_child_thread(void* arg)
+ {
+-	for (;;) ;
```

Comment:
do we need another child for infinite loop?

Reply:
Another child is not required as infinite loop is not required for this test.